src/features/Examples.ts

```diff
@@ -9,12 +10,16 @@ export class ExamplesFeature implements vscode.Disposable {
     private examplesPath: string;
 
     constructor() {
-        this.examplesPath = path.resolve(__dirname, "../../examples");
+        this.examplesPath = path.resolve(__dirname, "../examples");
         this.command = vscode.commands.registerCommand("PowerShell.OpenExamplesFolder", () => {
             vscode.commands.executeCommand(
                 "vscode.openFolder",
                 vscode.Uri.file(this.examplesPath),
                 true);
+
+            // Return existence of the path for testing. The `vscode.openFolder`
+            // command should do this, but doesn't (yet).
+            return fs.existsSync(this.examplesPath)
```

Comment:
consider using the `vscode.workspace.fs.stat` API instead because:
* It will return a Thenable so you don't have to use a `Sync` API for an async operation
* It will future proof you if the PowerShell extension ever has a Web extension
* You already have to make the Uri with `vscode.Uri.file` and pass it to `vscode.openFolder` so you can pass that in to `stat`

Reply:
if you don't wanna use that, then at least use the non-Sync exists and return a Promise.

Reply:
Ah! I went to use `fs.exists` (non-sync) and it was deprecated.

Reply:
@TylerLeonhardt Will a `throw` interrupt the user here?

> throws - {@@linkcode FileSystemError.FileNotFound FileNotFound} when uri doesn't exist.

Reply:
Yeah so you should try/catch it and return a boolean.

Reply:
This seems more complicated than it needs to be.

* The extension is already using the Node `fs` API in many places, so it's currently "future-proofed." Adding this here will not significantly impact the work to correct that.
* While it's synchronous, it's executed after the command is called, and it's simply checking the existence of a file that must exist and will have just been read, so it's in the hot path and not going to take up a significant amount of time.

The suggested alternative brings in exceptions to handle and code to transform the result of of that try/catch back to a boolean, all for what is just a sanity check. I appreciated the suggestion, but I don't think it worked out in practice.

Reply:
dropping the docs on `fs.exists()` which say to use `fs.stat()` or `fs.access()` instead.
https://nodejs.org/api/fs.html#fs_fs_exists_path_callback

you're call but I try to avoid the `fs` sync APIs. Esp since PowerShell folks occasionally open network drives (*shutters*) in vscode.

Reply:
It's not avoided at all...but one day I will fix all that.